magicarp/pipeline/harmless_base_pipeline.py

```diff
@@ -0,0 +1,48 @@
+import json
+import gzip
+from magicarp.pipeline import Pipeline, register_datapipeline
+
+@register_datapipeline
+class HarmlessBasePipeline(Pipeline):
+    def __init__(self, path):
+        self.path = path
+        self.data = []
+    
+    def __getitem__(self, idx):
+        with gzip.open(self.path) as f:
```

Comment:
File should not be opened on call __getitem__. This means we load the entire file from disk every time we access a single item. Should load it into memory in the __init__.